File: ToDoList/app/src/main/java/com/todolist/main/ListViewTaskAdapter.java
package com.todolist.main;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.CheckBox;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.QuickContactBadge;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.todolist.db.TaskTableHelper;
import com.todolist.models.Task;

import java.util.List;

public class ListViewTaskAdapter extends ArrayAdapter<Task> {
    TaskTableHelper taskTableHelper;
    public ListViewTaskAdapter(TaskTableHelper taskTableHelper, @NonNull Context context, @NonNull List<Task> tasks) {
        super(context, R.layout.row_list_view, tasks);
        this.taskTableHelper = taskTableHelper;
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        LayoutInflater layoutInflater = (LayoutInflater)this.getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View row = layoutInflater.inflate(R.layout.row_list_view, parent, false);
        TextView title = row.findViewById(R.id.textView);
        Task task = getItem(position);
        ImageButton btn = (ImageButton)row.findViewById(R.id.btnItemTaskListVIew);

        btn.setOnClickListener(new BtnClickTaskListener(task,taskTableHelper,(MainActivity)this.getContext()));
        title.setText(task.getDefinition());
        return row;
    }
}

File: Study/ToDoList/app/src/main/java/ru/brovkin/todolist/main/ListViewTaskAdapter.kt
package ru.brovkin.todolist.main

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageButton
import android.widget.TextView
import ru.brovkin.todolist.R
import ru.brovkin.todolist.models.Task
import ru.brovkin.todolist.repositories.TaskRepository

class ListViewTaskAdapter : ArrayAdapter<Task> {
    private lateinit var taskRepository: TaskRepository

    constructor(taskRepository : TaskRepository, context : Context, tasks : List<Task>) : super(context, R.layout.row_list_view, tasks) {
        this.taskRepository = taskRepository
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var layoutInflater = this.context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        var row = layoutInflater.inflate(R.layout.row_list_view, parent, false)
        var title = row.findViewById<TextView>(R.id.textView)
        var task = getItem(position)
        var btn = row.findViewById(R.id.btnItemTaskListVIew) as ImageButton

        btn.setOnClickListener(BtnClickTaskListener(task!!, taskRepository, this.getContext() as MainActivity))
        title.text = task.definition
        return row
    }
}